### PR TITLE
ref_gl: Fixed building with -fno-common (GCC 10)

### PR DESCRIPTION
**Pull request by FOTMarut**
- Added APIENTRY_LINKAGE define to specify linkage of GL functions in
  ref_gl/gl_export.h (i.e. auto, extern, static...)
  Defaults to extern
- Added appropriate defines to avoid duplicate definitions among
  different translation units

### CODE REVIEW

File: ref_soft/r_context.c [hunk 1]
```diff
@@ -17,14 +17,14 @@ GNU General Public License for more details.
 
 ref_api_t      gEngfuncs;
 ref_globals_t *gpGlobals;
-ref_instance_t RI;
+extern ref_instance_t RI;
```
Comment (a1batross) [inline ref_soft/r_context.c:20]: You can remove RI from here, it's defined in r_main.c and declared globally in r_local.h.

File: ref_soft/r_context.c [hunk 2]
```diff
@@ -17,14 +17,14 @@ GNU General Public License for more details.
...
 gl_globals_t tr;
 ref_speeds_t r_stats;
 byte *r_temppool;
 cvar_t *gl_emboss_scale;
-cvar_t *r_drawentities;
+extern cvar_t *r_drawentities;
```
Comment (a1batross) [inline ref_soft/r_context.c:25]: Same.

File: ref_soft/r_context.c [hunk 3]
```diff
@@ -17,14 +17,14 @@ GNU General Public License for more details.
...
 cvar_t *r_norefresh;
-cvar_t *vid_gamma;
+extern cvar_t *vid_gamma;
```
Comment (a1batross) [inline ref_soft/r_context.c:27]: Same.

File: ref_soft/r_bsp.c [hunk 4]
```diff
@@ -24,7 +24,7 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 //
 // current entity info
 //
-qboolean		insubmodel;
+extern qboolean		insubmodel;
```
Comment (a1batross) [inline ref_soft/r_bsp.c:27]: Same.

**Comment by a1batross** [comment]
I also recommend splitting this big commit to three. One where you just add APIENTRY_LINKAGE macro, second where you start to use it in ref_gl and third is fixes for ref_soft.

**Comment by FOTMarut** [comment]
Better?

**Comment by a1batross** [comment]
Yeah, great! Merging.

**Pull request merged**